popx/transaction_test.go

```diff
@@ -130,3 +143,28 @@ func (t popWriteSkewTest) UpdateBalance(
 	}
 	return nil
 }
+
+func collectCount(t *testing.T) (labelName, labelValue string, count int) {
+	var (
+		mChan   = make(chan prometheus.Metric)
+		metrics []prometheus.Metric
+		done    = make(chan struct{})
+	)
+	go func() {
+		defer close(done)
+		for m := range mChan {
+			metrics = append(metrics, m)
+		}
+	}()
+	TransactionRetries.Collect(mChan)
+	close(mChan)
+	<-done
+	for _, m := range metrics {
+		var pb dto.Metric
+		require.NoError(t, m.Write(&pb))
+		require.NotNil(t, pb.Counter)
+		require.NotEmpty(t, pb.Label)
+		return *pb.Label[0].Name, *pb.Label[0].Value, int(*pb.Counter.Value)
```

Comment:
Why return in the loop? If you are only interested in the first value you can just make this whole function A LOT easier. Also, in case `len(metrics) == 0` we would just silently return the zero values, but I guess the test should fail.

Reply:
Yeah, I got carried away a little while modifying the code from prometheus/testutil. Let's just say testing prometheus metrics is not exactly straightforward.

I can revisit the code if you want. Then again 🤷 

LMKWYT.